src/MainWindow.vala

```diff
@@ -817,4 +840,149 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
+                button_0.activate();
+                return true;
+            case Gdk.Key.@1:
+            case Gdk.Key.KP_1:
+                button_1.activate();
+                return true;
+            case Gdk.Key.@2:
+            case Gdk.Key.KP_2:
+                button_2.activate();
+                return true;
+            case Gdk.Key.@3:
+            case Gdk.Key.KP_3:
+                button_3.activate();
+                return true;
+            case Gdk.Key.@4:
+            case Gdk.Key.KP_4:
+                button_4.activate();
+                return true;
+            case Gdk.Key.@5:
+            case Gdk.Key.KP_5:
+                button_5.activate();
+                return true;
+            case Gdk.Key.@6:
+            case Gdk.Key.KP_6:
+                button_6.activate();
+                return true;
+            case Gdk.Key.@7:
+            case Gdk.Key.KP_7:
+                button_7.activate();
+                return true;
+            case Gdk.Key.@8:
+            case Gdk.Key.KP_8:
+                button_8.activate();
+                return true;
+            case Gdk.Key.@9:
+            case Gdk.Key.KP_9:
+                button_9.activate();
+                return true;
+            case Gdk.Key.plus:
+            case Gdk.Key.KP_Add:
+                button_add.activate();
+                return true;
+            case Gdk.Key.minus:
+            case Gdk.Key.KP_Subtract:
+                button_sub.activate();
+                return true;
+            case Gdk.Key.asterisk:
+            case Gdk.Key.KP_Multiply:
+                button_mult.activate();
+                return true;
+            case Gdk.Key.slash:
+            case Gdk.Key.KP_Divide:
+                button_div.activate();
+                return true;
+            case Gdk.Key.period:
+            case Gdk.Key.decimalpoint:
+            case Gdk.Key.KP_Decimal:
+                button_point.activate();
+                return true;
+            case Gdk.Key.BackSpace:
+            case Gdk.Key.KP_Delete:
+                button_del.activate();
+                return true;
+            case Gdk.Key.Return:
+            case Gdk.Key.KP_Enter:
+            case Gdk.Key.KP_Equal:
+                button_calc.activate();
+                return true;
+        }
+
+        switch (keyval) {
+            case Gdk.Key.percent:
+                button_percent.activate();
+                return true;
+            case Gdk.Key.parenleft:
+                button_par_left.activate();
+                return true;
+            case Gdk.Key.parenright:
+                button_par_right.activate();
+                return true;
+        }
+
+        if ((mod_state & Gdk.ModifierType.CONTROL_MASK) == Gdk.ModifierType.CONTROL_MASK) {
+            switch (keyval) {
+                case Gdk.Key.X:
+                case Gdk.Key.x:
+                    button_clr.activate();
+                    return true;
+                case Gdk.Key.e:
```

Comment:
I guess this needs to be approved by UX team. Gnome Calculator uses Ctrl+E for entering numbers in scientific format. I guess you can remove this and add this functionality later in a separate PR after UX team gives an approval or suggests an alternative solution.

Reply:
I agee